service/terraform-plugin/src/main/java/com/solace/maas/ep/event/management/agent/plugin/terraform/manager/TerraformUtils.java

```diff
@@ -70,30 +80,71 @@ public static Path createConfigPath(CommandRequest request, String directory) {
         return configPath;
     }
 
-    public static void writeHclToFile(Command command, Path configPath) throws IOException {
-        if (StringUtils.isNotEmpty(command.getBody())) {
-            // At the moment, we only support base64 decoding
-            Map<String, Object> parameters = command.getParameters();
-            if (parameters != null && parameters.containsKey(CONTENT_ENCODING) && "base64".equals(parameters.get(CONTENT_ENCODING))) {
-                byte[] decodedBytes;
-                try {
-                    decodedBytes = Base64.getDecoder().decode(command.getBody());
-                } catch (IllegalArgumentException e) {
-                    throw new IllegalArgumentException("Error decoding base64 content", e);
-                }
-                String filename = DEFAULT_TF_CONFIG_FILENAME;
-                if (parameters.containsKey("Output-File-Name")) {
-                    filename = (String) parameters.get("Output-File-Name");
-                }
-                Files.write(configPath.resolve(filename), decodedBytes);
-            } else {
-                if (parameters == null || !parameters.containsKey(CONTENT_ENCODING)) {
-                    throw new IllegalArgumentException("Missing Content-Encoding property in command parameters.");
-                }
+    public static Path createCommandExecutionLogDir(Path root) {
+        Path configPath = root.resolve("executionLogs");
+        if (Files.notExists(configPath)) {
+            try {
+                Files.createDirectories(configPath);
+            } catch (IOException e) {
+                throw new IllegalArgumentException(e);
+            }
+        }
+
+        return configPath;
+    }
+
+    public static void writeHclToFile(Command command,
+                                      Path configPath,
+                                      PrintWriter writer,
+                                      ObjectMapper objectMapper) {
+        try {
+            if (StringUtils.isNotEmpty(command.getBody())) {
+                // At the moment, we only support base64 decoding
+                Map<String, Object> parameters = command.getParameters();
+                if (parameters != null && parameters.containsKey(CONTENT_ENCODING) && "base64".equals(parameters.get(CONTENT_ENCODING))) {
+                    byte[] decodedBytes;
+                    try {
+                        decodedBytes = Base64.getDecoder().decode(command.getBody());
+                    } catch (IllegalArgumentException e) {
+                        throw new IllegalArgumentException("Error decoding base64 content", e);
```

Comment:
The caller is handling this exception. IMO, that is the best way forward. The caller can decide to either swallow the exception and continue processing, OR, it can fail fast